app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java

```diff
@@ -1792,18 +1906,78 @@ public void onClick(DialogInterface dialog, int which) {
             }
         });
 
+        FontTextView BtnSwitchSettings2 = findViewById(R.id.switchSettings2);
+        BtnSwitchSettings2.setTooltipText("Switch to Phone sensors mode");
+        BtnSwitchSettings2.setOnClickListener(new OnClickListener() {
+
+            @Override
+            public void onClick(View arg0) {
+
+                /*if (username == null || username.length() <1){
+                    Toast.makeText(ctx, getString(R.string.username_missing), Toast.LENGTH_LONG).show();
+                }else {
+
+                    //sensorManager.unregisterListener(MainActivity.this);
+                    Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
+                    MainActivity.this.startActivity(intent);
+                }*/
+                if (username == null || username.length() <1) {
+                    Toast.makeText(ctx, getString(R.string.username_missing), Toast.LENGTH_LONG).show();
+                }
+                else{
+                    SharedPreferences.Editor editor = sharedPreferences.edit();
+                    if(fullChartButton.getVisibility() == View.VISIBLE){
+                        hideCharts();
+                        editor.putString("dataTrackingSystem", getString(R.string.fitbit_tracking_ntt));
+                    }
+                    else{
+                        dayChart.setVisibility(View.VISIBLE);
+                        btnPieChart.setVisibility(View.VISIBLE);
+                        fullChartButton.setVisibility(View.VISIBLE);
+                        thirdPartyTracking.setVisibility(View.GONE);
+                        int steps = mStepsDBHelper.fetchTodayStepCount();
+                        displayActivityChart(steps, true);
+                        editor.putString("dataTrackingSystem", getString(R.string.device_tracking_ntt));
+                    }
+                    editor.apply();
+                }
+            }
+        });
+
+
+        BtnSwitchSettings.setTooltipText("Switch to Fitbit mode");

```

Comment:
The switch button will always be displayed in the app regardless of the mode. So having a fixed tooltip would be incorrect. Are you taking into consideration the user's current settings for loading the proper mode? the code for this should be in place. I don't think I've seen also any code that stores and saves the selection to the user's settings, or did i miss it? Scenario: user switches to fitbit mode. Closes the app, reopens the app. It should show in fitbit mode.

Reply:
there is 2 tool tip texts, one for each mode